liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/server/DevMojo.java

```diff
@@ -223,19 +227,205 @@ public void stopServer() {
             }
         }
         
+        @Override
+        public void startServer() {
+            try {
+                if (serverTask == null) {
+                    serverTask = initializeJava();
+                }
+                copyConfigFiles();
+                serverTask.setClean(clean);
+                serverTask.setOperation("start");
+                // Set server start timeout
+                if (serverStartTimeout < 0) {
+                    serverStartTimeout = 30;
+                }
+                serverTask.setTimeout(Long.toString(serverStartTimeout * 1000));
+                serverTask.execute();
+
+                if (verifyTimeout < 0) {
+                    verifyTimeout = 30;
+                }
+                long timeout = verifyTimeout * 1000;
+                long endTime = System.currentTimeMillis() + timeout;
+                if (applications != null) {
+                    String[] apps = applications.split("[,\\s]+");
+                    for (String archiveName : apps) {
+                        String startMessage = serverTask.waitForStringInLog(START_APP_MESSAGE_REGEXP + archiveName,
+                                timeout, serverTask.getLogFile());
+                        if (startMessage == null) {
+                            stopServer();
+                            throw new MojoExecutionException(MessageFormat
+                                    .format(messages.getString("error.server.start.verify"), verifyTimeout));
+                        }
+                        timeout = endTime - System.currentTimeMillis();
+                    }
+                }
+            } catch (Exception e) {
+                log.debug("Error starting server", e);
+            }
+        }
+        
+        @Override
+        public void getArtifacts(List<String> artifactPaths){
+            Set<Artifact> artifacts = project.getArtifacts();
+            for (Artifact artifact : artifacts) {
+                artifactPaths.add(artifact.getFile().getAbsolutePath());
+            }
+        }
+
+        @Override
+        public void recompileJava(List<File> javaFilesChanged, List<String> artifactPaths, ThreadPoolExecutor executor,
```

Comment:
Can this implementation be moved to ci.common as well?  Seems the only Maven specific part is getting the application name via `DevMojo.this.project.getArtifactId()`, but that can be passed in as a parameter to DevUtil.